src/civetweb.c

```diff
@@ -9449,10 +9481,22 @@ handle_request(struct mg_connection *conn)
 		}
 
 		/* 6. authorization check */
-		if (is_put_or_delete_request && !is_script_resource
-		    && !is_callback_resource) {
-/* 6.1. this request is a PUT/DELETE to a real file */
-/* 6.1.1. thus, the server must have real files */
+		/* 6.1. a custom authorization handler is installed */
+		if (get_request_handler(conn,
+		                        AUTH_HANDLER,
+		                        &auth_handler,
+		                        NULL,
+		                        NULL,
+		                        NULL,
+		                        NULL,
+		                        &auth_callback_data)) {
+			if (!auth_handler(conn, auth_callback_data)) {
```

Comment:
No, the type is checked. 'get_request_handler' in step 5 is called with WEBSOCKET_HANDLER or REQEUST_HANDLER. In step 6, 'get_request_handler' is called with AUTH_HANDLER.

The way I'm using it, maybe, 'get_request_handler' should be renamed 'get_handler'.